### PR TITLE
32-Bit Build Support

### PR DESCRIPTION
**Pull request by Ornamus**
**TLDR**:
- CrewLink now works on 32-bit Windows, installer installs 32-bit or 64-bit version as needed
- `iohook` files no longer need to be manually copied during setup

Looked into making working 32-bit builds of CrewLink after seeing issues #240 and #233. Added a `dist32` script that compiles a 32-bit .exe, and found that when launched it couldn't find the `iohook` module. It seemed this was because we were manually coping the 64-bit iohook binaries (with no 32-bit ones in sight) during initial project setup. 

To fix this, I added an iohook config to `package.json` that targets electron-80 for both 32-bit and 64-bit and let yarn take over fetching `iohook` instead of copying files manually - and it seems that did the trick! 64-bit builds still work fine, and 32-bit builds seem to launch and function fine too now.

**Note**: `yarn dist` and `yarn publish` build 32-bit and 64-bit versions, and will generate an installer that will install the appropriate version based off the user's architecture. If you want to build for a specific platform, use `yarn dist32` or `yarn dist64`.

### CODE REVIEW

**Comment by cinderblock** [comment]
@OiYouYeahYou Looks like you need to update `appveyor.yml` and remove the `- cp iohook ...` lines

**Comment by Ornamus** [comment]
@cinderblock Whoops, just realized I could've done that too. Removed them 👍

**Comment by Zaki360** [comment]
i still dont know how to set it up... pls help

File: package.json [hunk 1]
```diff
@@ -22,7 +22,20 @@
     "dev": "electron-webpack dev",
     "compile": "electron-webpack",
     "dist": "yarn compile && electron-builder --win --x64",
-    "publish": "yarn compile && electron-builder --win --x64 --publish always"
+    "dist32": "yarn compile && electron-builder --win --ia32",
```
Comment (cinderblock) [inline package.json:25]: I would just add `--ia32` to the regular `"dist"`, no?

Comment (Ornamus) [reply package.json:25]: That would work too, I just figured that if someone was using `dist` instead of `publish` that they'd likely be testing for their own system, and would not want the longer build time caused by building for both architectures. It's an easy enough change though if most people would rather have one longer `dist` command.

Comment (cinderblock) [reply package.json:25]: Or maybe add a `dist32` and `dist64` scripts?

Comment (Ornamus) [reply package.json:25]: Good idea! `dist32` already existed, so I just added `dist64` to match - and now `dist` targets both platforms, similar to `publish`.

**Comment by Ornamus** [comment]
I think I'm done making modifications, this should be ready to go 👍

**Comment by Yesitzme** [comment]
Can you please give us a date when the 32 bit version will come out?

**Comment by SADMaster04** [comment]
can someone make a video showing exactly what we have to do to get it on a 32bit pc please

**Comment by SADMaster04** [comment]
> I think I'm done making modifications, this should be ready to go 👍

please make a video tutorial!

**Comment by Ornamus** [comment]
> Can you please give us a date when the 32 bit version will come out?

There's no way to give a concrete date unfortunately. The feature is complete, I've tested that it works, and I've put it up for a review. Someone with write access to the repository will look over it at some point, and if they deem that it works and is up to the project's standards, it'll get merged in and become a part of the next CrewLink update.

> can someone make a video showing exactly what we have to do to get it on a 32bit pc please

There's not really a way to "get this" in the normal sense - it's a feature pull request, it's not actually part of CrewLink yet. If you follow the instructions on how to build the project, then pull this branch, you could build a 32 bit version yourself - but that's very much the path a developer would use to test this feature, not for an end user to do. I would just try to hold tight and wait for this to (hopefully) be merged in as a feature soon :)

**Comment by SADMaster04** [comment]
> > Can you please give us a date when the 32 bit version will come out?
> 
> There's no way to give a concrete date unfortunately. The feature is complete, I've tested that it works, and I've put it up for a review. Someone with write access to the repository will look over it at some point, and if they deem that it works and is up to the project's standards, it'll get merged in and become a part of the next CrewLink update.
> 
> > can someone make a video showing exactly what we have to do to get it on a 32bit pc please
> 
> There's not really a way to "get this" in the normal sense - it's a feature pull request, it's not actually part of CrewLink yet. If you follow the instructions on how to build the project, then pull this branch, you could build a 32 bit version yourself - but that's very much the path a developer would use to test this feature, not for an end user to do. I would just try to hold tight and wait for this to (hopefully) be merged in as a feature soon :)

well can u like explain how to build it?

**Comment by Ornamus** [comment]
> well can u like explain how to build it?

https://github.com/ottomated/CrewLink has the instructions, go to "Development" and read from there.

**Comment by cinderblock** [comment]
It looks like you could add a couple lines to `appveyor.yml` and the generated installer would be available as a build artifact on Appveyor. This could be done on this branch (or another), to satisfy the desires here for a 32-bit compatible version of the installer.

Something like:

```yml
...
artifacts:
  - path: dist/win-unpacked
    name: unpacked
    type: zip
  - path: 'dist/CrewLink Setup *.exe'
    name: CrewLink Setup Appveyor Build.exe
```

**Comment by SADMaster04** [comment]
does anyone know how to fix nat issues with 32bit windows7???

**Comment by Ornamus** [comment]
The changes from here got manually pulled into master, so I'll close this now. CrewLink versions 1.2.0 and later should now have 32-bit support :)

**Comment by Clemsko** [comment]
CrewLink-Setup-1.2.0.exe
